### PR TITLE
Bach/test/unit test 2

### PR DESCRIPTION
**Pull request by waridh**
This one is for the item class. It does testing for every public method on the Item class.

### CODE REVIEW

File: code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java [hunk 1]
```diff
@@ -342,13 +344,13 @@ public List<Tag> getTags() {
...
-     * Returns the list of tag names
+     * Returns the list of document tag ids.
      *
-     * @return
+     * @return a list of the document tag ID
      */
-    public ArrayList<String> getTagNames() {
-        return new ArrayList<>(tags.stream().map(Tag::getDocID).collect(Collectors
-                .toList()));
+    public ArrayList<String> getTagDocIDs() {
```
Comment (lizard-52) [inline code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java:351]: this is a bad name, it stores an array of strings. each string is used to rebuild a tag object from scratch (there is no second query to lookup each tag).

File: code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java [hunk 2]
```diff
@@ -342,13 +344,13 @@ public List<Tag> getTags() {
...
+        return tags.stream().map(Tag::getDocID).collect(Collectors
+                .toCollection(ArrayList::new));
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java:353]: faster construction because this version no longer construct twice.

**Comment by lizard-52** [comment]
For future testing PRs please try to keep the refactoring of existing code to a minimum.

File: code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java (hunk 2 above)
Comment (lizard-52) [reply code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java:353]: Assuming this is functionally the same I have no complaints.

**Review by lizard-52 — APPROVED** [review]
sure

**Review by echeng-git — APPROVED** [review]
I approve of these changes.

**Pull request merged**